File: cse240/hw14/question1.pl

/* RUle */
%foo is size n problem
foo(X,Y,Z) :- 
	Y =< 0, Z is X. %stop condition and its return value
foo(X,Y,Z) :- 
	X =< 0, Z is Y. %stop condition and its return value
foo(X,Y,Z) :- 
	X >= Y,                     %condtion check
	X1 is X - 1 , Y1 is Y - 2,  %size m problem
	foo(X1,Y1,Z1) , 	    %size m soulution
	Z = X + Z1.                 %output value
foo(X,Y,Z) :- 
	X < Y,  		    %condtion check
	X1 is X - 2 , Y1 is Y - 3,  %size m problem
	foo(X1,Y1,Z1) , 	    %size m soulution
	Z = Y + Z1.		    %output value

File: cse240/hw14/question1.cpp
#include <iostream>
using namespace std;
int foo(int x, int y)  //size n problem
{
	if (y <= 0)	      // stoping condtion 1
		return x;     // its return value
	else if (x <= 0)  // stoping condtion 2
		return y;     // its return value
	else if (x >= y)  // condtion check
		return x + foo(x - 1, y - 1);   // size m problem and its soulution
	else
		return y + foo(x - 2, y - 3);   // size m problem and its soultion 
}	
int main()
{
	cout << foo(5, 6);
	system("pause");
	return 0;
}